include/depthai_camera.h

```diff
@@ -0,0 +1,166 @@
+#ifndef FOG_SW_DEPTHAI_CAMERA_H
+#define FOG_SW_DEPTHAI_CAMERA_H
+#include <depthai/device/Device.hpp>
+#include <depthai/pipeline/datatype/ImgFrame.hpp>
+#include <depthai/pipeline/node/ColorCamera.hpp>
+#include <depthai/pipeline/node/MonoCamera.hpp>
+#include <depthai/pipeline/node/VideoEncoder.hpp>
+#include <depthai/pipeline/node/XLinkIn.hpp>
+#include <depthai/pipeline/node/XLinkOut.hpp>
+#include <depthai/utility/Initialization.hpp>
+#include <rclcpp/rclcpp.hpp>
+#include <sensor_msgs/msg/compressed_image.hpp>
+#include <sensor_msgs/msg/image.hpp>
+#include <std_msgs/msg/string.hpp>
+#include <iostream>
+
+namespace depthai_ctrl
+{
+
+/// ROS2 Node which works as DepthAI Camera driver
+/// Reads camera frames directly fromn the camera and publishes corresponding ROS2 messages
+/// It also publishes color camera video stream
+class DepthAICamera : public rclcpp::Node
+{
+  public:
+    using ImageMsg = sensor_msgs::msg::Image;
+    using CompressedImageMsg = sensor_msgs::msg::CompressedImage;
+
+    DepthAICamera()
+        : Node("depthai_camera"),
+          _videoWidth(1280),
+          _videoHeight(720),
+          _videoFps(25),
+          _videoBitrate(3000000),
+          _videoH265(false),
+          _thread_running(false),
+          _left_camera_frame("left_camera_frame"),
+          _right_camera_frame("right_camera_frame"),
+          _color_camera_frame("color_camera_frame")
```

Comment:
Should we use camera's functionality to create disparity data?

Reply:
there were no such requirement in the task, but It's easy to add.
However, personally I would not add it... because it's of low quality

Reply:
OK. No problem. Let's keep it as it is.